From 5cfb4ce27222d61e0fd85adc372edd98e1b92dea Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Date: Sat, 31 Oct 2015 22:48:07 +0100
Subject: tfm: make a few functions static

clamav expects them to be static and it does not seem bad to do so.

Patch-Name: tfm-make-a-few-functions-static.patch
Signed-off-by: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
---
 src/bin/fp_read_radix.c        | 2 +-
 src/bin/fp_read_signed_bin.c   | 2 +-
 src/bin/fp_read_unsigned_bin.c | 2 +-
 src/headers/tfm.h              | 6 +++---
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/bin/fp_read_radix.c b/src/bin/fp_read_radix.c
index cb1596c008ea..5bfe51982242 100644
--- a/src/bin/fp_read_radix.c
+++ b/src/bin/fp_read_radix.c
@@ -9,7 +9,7 @@
  */
 #include <tfm_private.h>
 
-int fp_read_radix(fp_int *a, char *str, int radix)
+int fp_read_radix(fp_int *a, const char *str, int radix)
 {
   int     y, neg;
   char    ch;
diff --git a/src/bin/fp_read_signed_bin.c b/src/bin/fp_read_signed_bin.c
index ebd4a3caf6ce..85eae66ea413 100644
--- a/src/bin/fp_read_signed_bin.c
+++ b/src/bin/fp_read_signed_bin.c
@@ -9,7 +9,7 @@
  */
 #include <tfm_private.h>
 
-void fp_read_signed_bin(fp_int *a, unsigned char *b, int c)
+void fp_read_signed_bin(fp_int *a, const unsigned char *b, int c)
 {
   /* read magnitude */
   fp_read_unsigned_bin (a, b + 1, c - 1);
diff --git a/src/bin/fp_read_unsigned_bin.c b/src/bin/fp_read_unsigned_bin.c
index cc4acb83e5df..59377dfdc3b8 100644
--- a/src/bin/fp_read_unsigned_bin.c
+++ b/src/bin/fp_read_unsigned_bin.c
@@ -9,7 +9,7 @@
  */
 #include <tfm_private.h>
 
-void fp_read_unsigned_bin(fp_int *a, unsigned char *b, int c)
+void fp_read_unsigned_bin(fp_int *a, const unsigned char *b, int c)
 {
   /* zero the int */
   fp_zero (a);
diff --git a/src/headers/tfm.h b/src/headers/tfm.h
index 436c94b2ea10..5ea43771b510 100644
--- a/src/headers/tfm.h
+++ b/src/headers/tfm.h
@@ -467,14 +467,14 @@ int fp_prime_random_ex(fp_int *a, int t, int size, int flags, tfm_prime_callback
 int fp_count_bits(fp_int *a);
 
 int fp_unsigned_bin_size(fp_int *a);
-void fp_read_unsigned_bin(fp_int *a, unsigned char *b, int c);
+void fp_read_unsigned_bin(fp_int *a, const unsigned char *b, int c);
 void fp_to_unsigned_bin(fp_int *a, unsigned char *b);
 
 int fp_signed_bin_size(fp_int *a);
-void fp_read_signed_bin(fp_int *a, unsigned char *b, int c);
+void fp_read_signed_bin(fp_int *a, const unsigned char *b, int c);
 void fp_to_signed_bin(fp_int *a, unsigned char *b);
 
-int fp_read_radix(fp_int *a, char *str, int radix);
+int fp_read_radix(fp_int *a, const char *str, int radix);
 
 int fp_radix_size(fp_int *a, int radix, int *size);
 int fp_toradix(fp_int *a, char *str, int radix);
